### ISSUE TITLE
A better way/method can be implemented to display the applied filter info.

### ISSUE THREAD

**Issue by Akshatji800** [CCExtractor/Flood_Mobile#115]
**Is your feature request related to a problem? Please describe.**
The current way of displaying the applied filter info in the search bar can be improved/changed to make it look better.

**Describe the solution you'd like**
To be proposed by the contributors who want to take up the issue by commenting on the issue.

**Comment by sainideepanshu199** [comment]
please assign this issue to me.

**Comment by sainideepanshu199** [comment]
I have installed Flood and qBittorrent but I am getting this error after following the contributing.md guide.
![image](https://user-images.githubusercontent.com/114583893/207404098-31ff565f-8b24-455f-b3c1-7a57dfbafceb.png)

**Comment by Akshatji800** [comment]
> I have installed Flood and qBittorrent but I am getting this error after following the contributing.md guide. ![image](https://user-images.githubusercontent.com/114583893/207404098-31ff565f-8b24-455f-b3c1-7a57dfbafceb.png)

@sainideepanshu199 there is some issue with qBittorent and flood which needs to be addressed and resolved until then you can use transmission instead of qBittorent. The setup is almost similar.

**Comment by Akshatji800** [comment]
@sainideepanshu199 is there any progress on this issue? 

Also @sainideepanshu199 were you able to solve the issue you were facing while setting up the project? If not then this is what you need to do to resolve the issue.

Uninstall the older version of flood u have by using the command `sudo npm uninstall -g flood` and install the latest but not released version by using the command `sudo npm i -g @jesec/flood`

**Comment by sainideepanshu199** [comment]
i am very sorry for no work on this issue, due to some serious problems couldn't work on anything. 
but now i have started the setup again and i get a new issue while accessing app after opening web client from transmission app.
![image](https://user-images.githubusercontent.com/114583893/213084487-55370b77-1187-49f6-903c-bbc3ef0f8b2e.png)

could u please resolve this problem

**Comment by amitamrutiya2210** [comment]
For solve this error, while installing you have to install this Web Interface Option.

![image](https://user-images.githubusercontent.com/91112485/213257874-92c313c3-5181-4abe-91a1-ecc958e2e892.png)

**Comment by sainideepanshu199** [comment]
can someone help me resolve this issue now, even after entering correct authentication details it shows failed login 
![image](https://user-images.githubusercontent.com/114583893/213359677-6187f65c-c50e-4be9-ada2-4b53bf3bb368.png)

**Comment by amitamrutiya2210** [comment]
You have to enter a details that you use to create flood web-ui account. If it give this login error then try to uninstall flood using `sudo npm uninstall -g flood` and again try to install `sudo npm i -g @jesec/flood` and create new account. after create new flood account try to login with that details.

**Comment by Akshatji800** [comment]
Unassigning this issue to @sainideepanshu199 as there was no progress reported on the issue

**Comment by Ronakkkkk** [comment]
Can you assign me this issue.

**Comment by Akshatji800** [comment]
> Can you assign me this issue.

Hello @Ronakkkkk please let me know if you still want to work on this issue - if yes then I will assign this issue to you.

**Comment by Roshaen** [comment]
Hey @Akshatji800 I would like to work on this issue
Can you assign me?

**Comment by tb-rules10** [comment]
Please assign this issue to me.

**Comment by Akshatji800** [comment]
> Please assign this issue to me.

Please let me know what is your idea and how you are gonna improve it before I assign this issue to you.

**Comment by tb-rules10** [comment]
I am thinking of combining Filters and Sort into a single bottom sheet and selected filters and sort by status can be seen below the search bar like this:-

<img width="208" alt="image" src="https://github.com/CCExtractor/Flood_Mobile/assets/58645688/677aed7f-a732-4508-8b07-24e2fa7eebb7">

**Comment by tb-rules10** [comment]
If you are okay with my idea, should I continue working on this?

**Comment by Akshatji800** [comment]
> I am thinking of combining Filters and Sort into a single bottom sheet and selected filters and sort by status can be seen below the search bar like this:-
> 
> <img alt="image" width="208" src="https://user-images.githubusercontent.com/58645688/271766782-677aed7f-a732-4508-8b07-24e2fa7eebb7.png">

Hey @tb-rules10, thanks for proposing this but the reason why this was not followed and approved initially itself is that we want to save as much space as we can on the main screen to show torrents. This is the reason why search bar, download speed etc are hidden and appears on top swipe. Please suggest something that aligns with our minimal screen space usage goal.

**Comment by MrBooring** [comment]
Hello, I'm interested in addressing this issue, but I'd like to first explore the application to better understand it. Others have faced installation problems before. I'll return with a proposal once I've investigated it.
Thank you.

**Comment by Nailsonseat** [comment]
Hi @Akshatji800 its Aadarsh from FOSS 24. I think we can use chips to better portray the filtering options instead of the current radio selector. This proposal is also in line with using minimum space possible, since a user would have to scroll less.

Below is a skeleton of what the filter would look like, as the tracker list would be varying in size, the filter dialog will be scroll-able and will use wrap widget to handle devices of varying width.

<table>
  <tr>
    <td><img width="280" src="https://github.com/CCExtractor/Flood_Mobile/assets/67797628/4dae666b-68d1-47c6-b768-02d83b670780" alt="flood"></td>
  </tr>
</table>

**Comment by Akshatji800** [comment]
Hey @Nailsonseat assigning this issue to you. But this issue is more about how we are gonna show the current filter selected in a better way. Currently it is shown in searchbar. I want you to work on that first.

**Comment by Nailsonseat** [comment]
> Hey @Nailsonseat assigning this issue to you. But this issue is more about how we are gonna show the current filter selected in a better way. Currently it is shown in searchbar. I want you to work on that first.

Thank you Akshat, I will come back to you with an idea regarding this very soon.

**Comment by Nailsonseat** [comment]
@Akshatji800 I have three points regarding this problem

- I think displaying the filter in the search bar itself is a good idea and should stay that way if we intend to minimize space.
- When any filter is selected, a dismiss filter option would be a great addition instead of a placeholder filter icon.
- When no filter is selected displaying any text in the filter is redundant and should be limited to filter icon only



<img width="350" alt="Flood_1" src="https://github.com/CCExtractor/Flood_Mobile/assets/67797628/3e84a62e-2c69-4d45-8a0e-b9f447d3bd22">

<img width="350" alt="Flood_2" src="https://github.com/CCExtractor/Flood_Mobile/assets/67797628/685e6074-5df1-482d-a6ec-d2dbe3eaef11">

**Comment by Akshatji800** [comment]
@Nailsonseat good observation but don't you think that removing filter icon and putting a dismiss button instead of it creates confusion that what actually complete is? The idea is good that there should be a way to dismiss the filter (it is definitely increasing the UX) but you need to think of a better UI.

**Comment by Nailsonseat** [comment]
> @Nailsonseat good observation but don't you think that removing filter icon and putting a dismiss button instead of it creates confusion that what actually complete is? The idea is good that there should be a way to dismiss the filter (it is definitely increasing the UX) but you need to think of a better UI.

Good point, the user will not know if 'complete' is a filter or something else, will work on it. I deleted the original post because I wanted to show a good way to show sorting too along with filters. I will come to you will a new idea for both soon

**Comment by Nailsonseat** [comment]
@Akshatji800 I have some new mock-ups or rather old ones because these designs are more closer to the original UI.


<table>
<tr>
<td>Initial state</td>
<td>Selected state</td>
</tr>
  <tr>
    <td>
      <img width="400" height="350" alt="Flood_5" src="https://github.com/CCExtractor/Flood_Mobile/assets/67797628/5100ed21-9f4c-4573-b481-3530522abfec">
    </td>
    <td>
      <img width="400" height="350" alt="Flood_6" src="https://github.com/CCExtractor/Flood_Mobile/assets/67797628/5e272532-b3e1-4c50-8c89-4d6df00a4cfc">
    </td>
  </tr>
</table>

### Filters
I have two points with respect to the concern with confusion with dismiss button

- The current mock-up has UI which is more intuitive in terms of deducing what the "complete" and the X button stands for, IMO.
- The filter is located on far right where there is no other button near it, so I think the case for confusion would be even lower.


### Sort

- I included the sort button in a border to make it seem more like a button but I'm not sure about it, need you input on this.
- Dismiss button is not needed for sorting as there is always a sort type that is always selected.
- Currently whichever sort type you select , it always displays the A-Z icon, I think we can find all the 12 icons for the 12 sort types of sorts [here](https://fonts.google.com/icons)


Do let me know if more improvements can be made here

**Comment by Nailsonseat** [comment]
i found a potential implementation for the sort button straight from the material 3 website. Just imagine this with icons, and without the text in the button. The focus is the button not the drop down itself


https://github.com/CCExtractor/Flood_Mobile/assets/67797628/22a7d132-7993-451a-8df8-10439b99beab


I ll be on the lookout for better options
